src/modals/SendModal.js

```diff
@@ -427,9 +442,26 @@ class SendModal extends Component {
                   value={this.props.recipient}
                   onFocus={this.onAddressInputFocus}
                   onBlur={this.onAddressInputBlur}
-                  onChange={({ target }) =>
-                    this.props.sendUpdateRecipient(target.value)
-                  }
+                  onChange={({ target }) => {
+                    if (target.value.includes('.')) {
+                      ens
+                        .lookup(target.value.trim())
+                        .then(address => {
+                          this.props.sendUpdateRecipient(
+                            address,
+                            this.props.selected.symbol,
+                          );
+                        })
+                        .catch(reason => {
+                          console.log('ENS failed');
+                          // console.error(reason);
+                        });
+                    }
+                    this.props.sendUpdateRecipient(
+                      target.value.trim(),
+                      this.props.selected.symbol,
+                    );
+                  }}
```

Comment:
3. This logic should be outside of render() to make the code more legible
4. We need to allow the user to view the ens address using the `sendUpdateRecipient` will change the input

Reply:
Moved the logic to a separate function, removed the value attr to view the ens.